### DISCUSSION TITLE
Jedis: Timeout waiting for idle object

### DISCUSSION THREAD

**Discussion by yourzbuddha** [xetorthio/jedis#3820] [Q&A] [answered]
I'm getting `java.util.NoSuchElementException: Timeout waiting for idle` object during the init of the JedisPool with such configs:

> Specs: Scala: 2.12, jedis version: 3.10

``` scala
  private val jedisPoolConfig: JedisPoolConfig = {
    val config = new JedisPoolConfig()
    config.setMaxTotal(poolConfig.maxTotal)
    config.setMaxIdle(poolConfig.maxIdle)
    config.setMinIdle(4)
    config.setMaxWaitMillis(2000)
    config
  }
val pool = new JedisPool(jedisPoolConfig, redisHost, redisPort, 5000)
```
The error is from `borrowObject`, which is called on the pool during `pool.getResource`, because there are never available **idle** objects in the pool (it's created empty).

Is there a way to understand why it happens? Logging is not available, AFAIK.
I've checked Redis instance and it's healthy, up and running
 I really appreciate any help you can provide.

**Comment by sazzad16** [comment, marked as answer]
You can do `config.setBlockWhenExhausted(false)` to avoid that specific exception.

It's coming from commons-pool (version 2) library. You may check that library to learn more.
